### PR TITLE
creating CMakeLists.txt for low_voltage_signals_handler_test

### PR DESCRIPTION
**Pull request by SKobziev**
This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
These changes include low_voltage_signals_handler_test into a build. CMakeLists.txt for creating this test was refactored according to the changes of BSolonenko.

##### Breaking Changes
- included the folder "appMain/test" into the build
- changed appropriate CMakeLists.txt 
- added some changes into transport_manager/CMakeLists.txt to fix build errors
- changed function in sources.cmake "create test"

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by SKobziev** [comment]
@LuxoftAKutsan  @BSolonenko  Could you please review this PR?

File: src/appMain/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+project(low_voltage_signals_handler_test)
+
+create_test(${PROJECT_NAME})
+
+target_sources(${PROJECT_NAME}
+  PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/../low_voltage_signals_handler.cc
```
Comment (LuxoftAKutsan) [inline src/appMain/test/CMakeLists.txt:37]: @BSolonenko that is the place where low_voltage_signals_handler.cc library may be used. Please take in mind

File: src/appMain/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  ${CMAKE_CURRENT_LIST_DIR}/low_voltage_signals_handler_test.cc
+  ${COMPONENTS_DIR}/utils/test/mock_signals_posix.cc
+
+)
+
+target_link_libraries(${PROJECT_NAME}
+  PRIVATE
+  components::interface
+)
+
```
Comment (LuxoftAKutsan) [inline src/appMain/test/CMakeLists.txt:47]: @BSolonenko  remove empty lines

File: src/components/transport_manager/CMakeLists.txt [hunk 3]
```diff
@@ -53,6 +53,13 @@ target_sources(${PROJECT_NAME}
     "${CMAKE_CURRENT_LIST_DIR}/src/transport_adapter/transport_adapter_listener_impl.cc"
 )
 
+if(BUILD_TESTS)
+  target_sources(${PROJECT_NAME}
+    PRIVATE
+      "${CMAKE_CURRENT_LIST_DIR}/src/iap2_emulation/iap2_transport_adapter.cc"
```
Comment (LuxoftAKutsan) [inline src/components/transport_manager/CMakeLists.txt:59]: @SKobziev @BSolonenko  not clear,  why do we add additional sources within tests to the project?

File: tools/cmake/helpers/sources.cmake [hunk 4]
```diff
@@ -134,12 +134,33 @@ function(collect_sources SOURCES PATHS)
...
 
-function(create_test NAME SOURCES LIBS)
-  add_executable("${NAME}" ${CMAKE_SOURCE_DIR}/src/components/test_main.cc ${SOURCES})
-  target_link_libraries("${NAME}" ${LIBS})
-  target_link_libraries("${NAME}" Utils)
-  add_test(NAME ${NAME}
-  COMMAND ${NAME} --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/)
+function(create_test NAME)
+
+add_executable(${NAME})
```
Comment (LuxoftAKutsan) [inline tools/cmake/helpers/sources.cmake:139]: @SKobziev @BSolonenko  fix indentation

File: tools/cmake/helpers/sources.cmake [hunk 5]
```diff
@@ -134,12 +134,33 @@ function(collect_sources SOURCES PATHS)
...
+    ${GMOCK_INCLUDE_DIRECTORY}
+    "${COMPONENTS_DIR}/utils/test/include"
+)
+
+target_sources(${NAME}
+  PRIVATE
+    ${CMAKE_SOURCE_DIR}/src/components/test_main.cc
+)
+
+find_package(Boost 1.66 COMPONENTS filesystem system thread REQUIRED)
```
Comment (LuxoftAKutsan) [inline tools/cmake/helpers/sources.cmake:153]: @BSolonenko @SKobziev  use proper way to find boost if it is not on the system

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

File: src/components/transport_manager/CMakeLists.txt (hunk 3 above)
Comment (BSolonenko) [reply src/components/transport_manager/CMakeLists.txt:59]: @LuxoftAKutsan We have `include` that depend on BUILD_TESTS
https://github.com/BSolonenko/sdl_core/blob/df9b67026d90f0f9cdf04bec7dd3c7666a25db87/src/components/transport_manager/src/transport_manager_default.cc#L47-L49

**Pull request closed without merging**